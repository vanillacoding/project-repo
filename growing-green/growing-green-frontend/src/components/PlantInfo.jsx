import React from 'react';
import styled from 'styled-components';

export default function PlantInfo({ plant }) {
  return (
    <Container>
      <h3 className="info-item">식물 정보</h3>
      <p className="info-item">이름: {plant.name}</p>
      <p className="info-item">학명: {plant.scientificName}</p>
      <p className="info-item">과: {plant.species}</p>
      <p className="info-item">물주기: {plant.watering}일</p>
      <p className="info-item">
        광도: {plant.isSunPlant === true ? '양지 식물' : '음지식물'}
      </p>
    </Container>
  );
}

const Container = styled.div`
  text-align: left;
  margin: 2rem 2rem 0 -2rem;
  position: relative;
  border: 1px solid ${({ theme }) => theme.baseTheme.colors.gray};
  background: ${({ theme }) => theme.baseTheme.colors.ivory};
  padding: 1rem;

  .info-item {
    margin-left: 2rem;
  }
`;
