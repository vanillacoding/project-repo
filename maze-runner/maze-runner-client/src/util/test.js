import { NODE_STATUS } from '../constant';
import { createNodes } from './maze';

export function getNodeColIndex(nodeId) {
  if (typeof nodeId !== 'string' && !(nodeId instanceof String)) {
    throw new Error('Node id should be a string');
  }

  if (nodeId.split('-').length !== 2) {
    throw new Error('Invalid node id');
  }

  return nodeId.split('-')[1];
}

export function getNodeRowIndex(nodeId) {
  if (typeof nodeId !== 'string' && !(nodeId instanceof String)) {
    throw new Error('Node id should be a string');
  }

  if (nodeId.split('-').length !== 2) {
    throw new Error('Invalid node id');
  }

  return nodeId.split('-')[0];
}

export function getPrevNodeId(nodeId) {
  if (typeof nodeId !== 'string' && !(nodeId instanceof String)) {
    throw new Error('Node id should be a string');
  }

  if (nodeId.split('-').length < 2) {
    throw new Error('Invalid node id');
  }

  const row = getNodeRowIndex(nodeId);
  const col = getNodeColIndex(nodeId);

  return `${Number(row)}-${Number(col) - 1}`;
}

export function createById(startNodeId, endNodeId) {
  const exampleBlocks = [
    [0, 0, 0, 1, 0, 0, 1, 0, 0, 1],
    [0, 1, 0, 1, 0, 1, 1, 0, 1, 1],
    [0, 1, 0, 0, 0, 0, 0, 0, 0, 0],
    [1, 1, 0, 1, 1, 1, 1, 1, 1, 1],
    [0, 0, 0, 1, 0, 1, 1, 1, 0, 0],
    [0, 1, 1, 1, 0, 0, 0, 0, 0, 0],
    [0, 0, 0, 0, 1, 1, 1, 1, 1, 0],
    [1, 1, 1, 0, 1, 0, 0, 0, 0, 0],
    [1, 0, 0, 0, 1, 0, 1, 1, 0, 1],
    [1, 0, 1, 0, 0, 0, 0, 1, 0, 0],
  ];

  const startNodeIndex = {
    col: getNodeColIndex(startNodeId),
    row: getNodeRowIndex(startNodeId),
  };
  const endNodeIndex = {
    col: getNodeColIndex(endNodeId),
    row: getNodeRowIndex(endNodeId),
  };

  exampleBlocks[startNodeIndex.row][startNodeIndex.col] = 3;
  exampleBlocks[endNodeIndex.row][endNodeIndex.col] = 4;

  const [widthCount, heightCount] = [
    exampleBlocks[0].length,
    exampleBlocks.length,
  ];

  const nodes = createNodes(widthCount, heightCount);
  const { byId } = nodes;

  exampleBlocks.forEach((row, rowIndex) =>
    row.forEach((number, colIndex) => {
      const nodeId = `${rowIndex}-${colIndex}`;

      switch (number) {
        case 0: {
          byId[nodeId].status = NODE_STATUS.UNVISITED;
          return;
        }
        case 1: {
          byId[nodeId].status = NODE_STATUS.WALL;
          return;
        }
        case 2: {
          byId[nodeId].status = NODE_STATUS.WEIGHTED;
          byId[nodeId].weight = 10;
          return;
        }
        case 3: {
          byId[nodeId].status = NODE_STATUS.START;
          return;
        }
        case 4: {
          byId[nodeId].status = NODE_STATUS.END;
          return;
        }
        case 5: {
          byId[nodeId].status = NODE_STATUS.MIDDLE;
          return;
        }
        default: {
          return;
        }
      }
    }),
  );

  return byId;
}

export default {
  getNodeColIndex,
  getNodeRowIndex,
  getPrevNodeId,
  createById,
};
