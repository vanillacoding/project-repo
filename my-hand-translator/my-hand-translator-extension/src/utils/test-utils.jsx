import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";

import userReducer from "../features/user/userSlice";

function render(
  ui,
  {
    preloadedState,
    store = configureStore({ reducer: { user: userReducer }, preloadedState }),
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";

const user = {
  projectId: "translate-324702",
  translations: [
    {
      nanoId: "qWYBIfLPMsjP1LHNRqlFW",
      origin: "react",
      translated: "리액트",
      url: "vanilla",
      glossary: {
        react: "리액트",
      },
      createdAt: "2017-01-01",
    },
  ],
  glossary: {
    react: "리액트",
  },
  glossaryId: "6141bd1c2a11e5b51320bac1",
  isServerOn: false,
  name: "aidencoders",
  email: "aidencoders@gmail.com",
};

export { render, user };
