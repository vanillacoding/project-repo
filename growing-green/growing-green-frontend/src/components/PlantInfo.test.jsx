import React from 'react';
import PlantInfo from './PlantInfo';
import { render } from '../test/test-utils';

describe('<PlantInfo>', () => {
  const mockPlant = {
    name: '장미',
    scientificName: 'Rosaceae 로사케아이',
    species: '장미과',
    watering: 5,
    isSunPlant: true,
  };

  it('matches snapshot', () => {
    const { container } = render(<PlantInfo plant={mockPlant} />);
    expect(container).toMatchSnapshot();
  });

  it('should render plant information', () => {
    const { getByText } = render(<PlantInfo plant={mockPlant} />);
    expect(getByText(/식물 정보/i)).toBeInTheDocument();
    expect(getByText(`이름: ${mockPlant.name}`)).toBeInTheDocument();
    expect(getByText(`학명: ${mockPlant.scientificName}`)).toBeInTheDocument();
    expect(getByText(`과: ${mockPlant.species}`)).toBeInTheDocument();
    expect(getByText(`물주기: ${mockPlant.watering}일`)).toBeInTheDocument();
    expect(getByText(/양지 식물/i)).toBeInTheDocument();
  });
});
