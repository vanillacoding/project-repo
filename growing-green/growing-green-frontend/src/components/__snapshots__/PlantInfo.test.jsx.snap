// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlantInfo> matches snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL fMRptz"
  >
    <h3
      class="info-item"
    >
      식물 정보
    </h3>
    <p
      class="info-item"
    >
      이름: 
      장미
    </p>
    <p
      class="info-item"
    >
      학명: 
      Rosaceae 로사케아이
    </p>
    <p
      class="info-item"
    >
      과: 
      장미과
    </p>
    <p
      class="info-item"
    >
      물주기: 
      5
      일
    </p>
    <p
      class="info-item"
    >
      광도: 
      양지 식물
    </p>
  </div>
</div>
`;
